src/purchase_listinator/dbs/datomic/purchase_category.clj

```diff
@@ -0,0 +1,48 @@
+(ns purchase-listinator.dbs.datomic.purchase-category
+  (:require [schema.core :as s]
+            [datomic.api :as d]
+            [purchase-listinator.adapters.db.purchase-category :as adapters.db.purchase-category]))
+
+(def schema
+  [{:db/ident       :purchase-category/id
+    :db/valueType   :db.type/uuid
+    :db/cardinality :db.cardinality/one
+    :db/unique      :db.unique/identity
+    :db/doc         "The purchase-category id"}
+   {:db/ident       :purchase-category/name
+    :db/valueType   :db.type/string
+    :db/cardinality :db.cardinality/one
+    :db/doc         "The purchase-category name"}
+   {:db/ident       :purchase-category/order-position
+    :db/valueType   :db.type/long
+    :db/cardinality :db.cardinality/one
+    :db/doc         "The purchase-category order postion"}
+   {:db/ident       :purchase-category/color
+    :db/valueType   :db.type/string
+    :db/cardinality :db.cardinality/one
+    :db/doc         "The purchase-category color"}
+   {:db/ident       :purchase-category/purchase-list
+    :db/cardinality :db.cardinality/one
+    :db/valueType   :db.type/ref}])
+
+(s/defn ^:private transact
+  [connection & purchases-categories]
+  @(d/transact connection purchases-categories))
+
+(s/defn get-by-id
```

Comment:
I think this function could be separated in two, one for a generic get-by-id and other that inform the id key.

I don't know if the datomic allows it but I think I could give it a try

Reply:
Other PR